tests/vm_delete_protection_test.go

```diff
@@ -61,6 +61,8 @@ var _ = Describe("VM delete protection", func() {
 		vm = createVMWithDeleteProtection(labelValue)
 
 		Expect(apiClient.Delete(ctx, vm)).To(Succeed())
+		waitForDeletion(client.ObjectKeyFromObject(vm), &kubevirtv1.VirtualMachine{})
+		vm = nil
```

Comment:
Without this line, it should still work if we keep `MatchError(errors.IsNotFound, "errors.IsNotFound")` in the cleanup.

Reply:
Agree, let's try.